pyproject.toml

```diff
@@ -18,7 +18,10 @@ readme = {file = "README.md", content-type = "text/markdown"}
 license = {text = "MIT"}
 requires-python = ">=3.10"
 dependencies = [
+  "IPython",
```

Comment:
This should be an optional dependency. Check dynamically for the existence of `IPython` at runtime.